### PR TITLE
Apache Spark Example extended to use GlusterFS Volume Plugin

### PR DESCRIPTION
**Pull request by wattsteve**
I've extended the original v1.0 Spark Example to demonstrate how Spark can run analytics on a variety of network storage providers by integrating Kubernetes Volume Plugins, using the GlusterFS Volume Plugin as an exemplar. Spark is most commonly used with a Distributed File System and the current example isn't particularly useful without this type of integration.

### CODE REVIEW

File: examples/spark/spark-gluster/glusterfs-endpoints.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (mikedanese) [inline examples/spark/spark-gluster/glusterfs-endpoints.json]: .json files should be .yaml

File: examples/spark/spark-gluster/spark-worker-rc.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      },
+      "spec": {
+        "containers": [
+          {
+            "name": "spark-worker",
+            "image": "gcr.io/google_containers/spark-worker",
+            "privileged": true,
+            "ports": [
+              {
+                "hostPort": 8888,
```
Comment (mikedanese) [inline examples/spark/spark-gluster/spark-worker-rc.json]: is hostPort still required? why can't spark use the pod's IP and containerPort?

**Comment by mikedanese** [comment]
you need to run ./hack/run-gendocs.sh and fix the test in https://github.com/GoogleCloudPlatform/kubernetes/blob/master/examples/examples_test.go

**Comment by vishh** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
Please reopen when you have time to fix this up to pass precommit checks.

**Comment by jonKSA** [comment]
@jayunit100 Starting with a fresh clone of kubernetes repo, running `go run hack/e2e.go -v -build -up -test` results in this error:

...
Step 3 : RUN cd /usr/src/go/src && for platform in ${KUBE_CROSSPLATFORMS}; do GOOS=${platform%/*} GOARCH=${platform##*/} ./make.bash --no-clean; done
 ---> Running in f14605335123
unexpected EOF
...
!!! Error in /root/kubernetes/hack/e2e-internal/../../cluster/vagrant/../../cluster/vagrant/../../cluster/../build/../build/common.sh:369
  'return 1' exited with status 1

This also causes the docker daemon to crash.  journalctl -eu docker doesn't log any error message.

**Comment by jayunit100** [comment]
@jonksa next ste
- clone steve's fork of kube
- switch to his branch (git checkout wattsteve/master) 
- run local-up make sure it works
- run hack/verify-gofmt, make sure formatting is ok 
- use a  json->yaml conversion utility from somewhere ( http://jsontoyaml.com/ ), and see if the individual files still work with `cluster/kubectl.sh create -f blah.yaml`

**Comment by jayunit100** [comment]
MOVING THIS PR.

Ok lets issue a brand new PR... it will be easy that way.

-  Steve's commits will be in your branch (https://github.com/copejon/kubernetes)
- PR direct from copejon -> kubernetes master.  Make sure steve's as well as your commits are preserved !  

Looking forward to this example for our integration tests.  Thanks!

**Comment by copejon** [comment]
@wattsteve @jayunit100 The re issued PR can be found at:
https://github.com/kubernetes/kubernetes/pull/13109

**Pull request closed without merging**